Formula/openmama.rb

```diff
@@ -0,0 +1,47 @@
+class Openmama < Formula
+  desc "Open source high performance messaging API for various Market Data sources"
+  homepage "https://openmama.org"
+  url "https://github.com/finos/OpenMAMA/archive/OpenMAMA-6.3.1-release.tar.gz"
+  sha256 "43e55db00290bc6296358c72e97250561ed4a4bb3961c1474f0876b81ecb6cf9"
+  license "LGPL-2.1-only"
+
+  depends_on "cmake" => :build
+  depends_on "apr"
+  depends_on "apr-util"
+  depends_on "libevent"
+  depends_on "ossp-uuid"
+  depends_on "qpid-proton"
+  uses_from_macos "flex" => :build
+
+  def install
+    mkdir "build" do
+      system "cmake", "..",
+        "-DAPR_ROOT=#{Formula["apr"].opt_prefix}",
+        "-DPROTON_ROOT=#{Formula["qpid-proton"].opt_prefix}",
+        "-DINSTALL_RUNTIME_DEPENDENCIES=OFF",
+        "-DWITH_TESTTOOLS=OFF",
+        *std_cmake_args
+      system "make", "install"
+    end
+  end
+
+  test do
+    system "#{bin}/mamalistenc", "-?"
+    (testpath/"test.c").write <<~EOS
+      #include <mama/mama.h>
+      #include <stdio.h>
+      int main() {
+          mamaBridge bridge;
+          fclose(stderr);
+          mama_status status = mama_loadBridge(&bridge, \"qpid\");
+          if (status != MAMA_STATUS_OK) return 1;
+          const char* version = mama_getVersion(bridge);
+          if (NULL == version) return 2;
+          printf(\"%s\\n\", version);
+          return 0;
+      }
+    EOS
+    system ENV.cc, "test.c", "-I#{include}", "-L#{lib}", "-lmama", "-Wl,-rpath,#{lib}", "-o", "test"
```

Comment:
Having to set `-rpath` here is troubling. This should already be set correctly at build-time. What happens if you drop the `-rpath` flag?

Reply:
It will compile but won't run (I had to rent an arm server just to find out)... it's weird because it is not required for non-arm.

Reply:
>I had to rent an arm server just to find out

I appreciate your dedication to this PR, but I personally wouldn't have minded if you had used our CI to test it, especially since the CI run here doesn't take too long.

Try adding the flag
```
-DCMAKE_INSTALL_RPATH=#{opt_lib}
```
to the CMake build instead. If the libraries have the wrong `rpath`, then it's likely that the executables do as well, which is a problem for normal use on ARM.

Reply:
Yes thanks I'll try that though it did bother me why this behaviour was different in x64 vs arm and it looks like this is down to the homebrew installation of paths and symlinks being different. Not sure how old the original brew version I have is, but on x64, the libraries seem to get symlinked by default to `/usr/local/lib/*`:

```
osx:~ fquinn$ ls -l /usr/local/lib/libmama.dylib
lrwxr-xr-x  1 fquinn  admin  42 Mar  1 14:27 /usr/local/lib/libmama.dylib -> ../Cellar/openmama/6.3.1/lib/libmama.dylib
```

Which based on `ld` (I'm a linux guy, not sure if thats even up to date on OSX any more) is included on the search path on x64 but *not* on arm. Not sure if that is something that I happen to have done which your CI servers also happen to have done but neverthess that's what my system reports. On arm, there is nothing in /usr/local/lib - everything is in /opt/homebrew/lib only (which makes sense to avoid conflicts tbh).

The funny thing is that if the test that I put on was just a `--version` that didn't explicitly attempt to dlopen a plugin (dynamic linked dependencies already loaded fine), I wouldn't have noticed that this was an issue at all, so I expect I won't be the only one to get stumped by this.

Anyhow thanks for the advice, I'll try with explicit RPATH setting in the mama library itself.

Reply:
Seemed to work ok in my test env... just waiting on CI now.

Reply:
Yup looks good now!

Reply:
Great! GitHub Actions isn't working properly at the moment, so it'll take some time to merge this. In the meantime, please squash your commits together with your first one.

Thanks!

Reply:
Done and thanks again!